flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/typeutils/BaseArraySerializer.java

```diff
@@ -134,6 +139,48 @@ private GenericArray copyGenericArray(GenericArray array) {
 		return new GenericArray(arr, array.numElements(), array.isPrimitiveArray());
 	}
 
+	private GenericArray copyColumnarArray(ColumnarArray from) {
+		Object arr;
+		boolean isPrimitiveArray = true;
+		switch (eleType.getTypeRoot()) {
+			case BOOLEAN:
+				arr = from.toBooleanArray();
```

Comment:
We can invoke toPrimitiveArray only then nullable is false?

Reply:
Good catch. We should check the nullable. Updated the PR.